### PR TITLE
Allow uploading of plain text bindings

### LINKED ISSUE #993: include plain text bindings in upload form
when we wrangler publish a project with plain text bindings, we need to include the bindings in the upload form metadata.json.

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
fixes #993

### CODE REVIEW

File: src/commands/publish/upload_form/mod.rs [hunk 1]
```diff
@@ -132,6 +157,8 @@ fn add_metadata(mut form: Form, assets: &ProjectAssets) -> Result<Form, failure:
         bindings: assets.bindings(),
     });
 
+    println!("{}", metadata_json);
```
Comment (ashleymichal) [inline src/commands/publish/upload_form/mod.rs]: leftover println?

File: src/settings/binding.rs [hunk 2]
```diff
@@ -2,21 +2,17 @@ use serde::Serialize;
 
 #[derive(Serialize, Debug)]
 #[serde(tag = "type")]
+#[serde(rename_all = "snake_case")]
```
Comment (ashleymichal) [inline src/settings/binding.rs]: nice

File: src/settings/toml/manifest.rs [hunk 3]
```diff
@@ -198,9 +200,17 @@ impl Manifest {
...
-            if environment.webpack_config.is_some() {
-                target.webpack_config = environment.webpack_config.clone();
+            if let Some(webpack_config) = &environment.webpack_config {
+                target.webpack_config = Some(webpack_config.clone());
             }
+            if let Some(target_config) = &mut target.config.clone() {
+                if let Some(env_config) = environment.config.clone() {
+                    target_config.extend(env_config);
+                    let new_config = target_config.clone();
+                    target.config.replace(new_config);
```
Comment (ashleymichal) [inline src/settings/toml/manifest.rs]: 👍

File: src/commands/publish/upload_form/plain_text.rs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+use super::binding::Binding;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PlainText {
+    pub name: String,
+    pub value: String,
```
Comment (victoriabernard92) [inline src/commands/publish/upload_form/plain_text.rs]: thought we called this `text` not value? @ashleymichal

Comment (ashleymichal) [reply src/commands/publish/upload_form/plain_text.rs]: this is where the binding is built https://github.com/cloudflare/wrangler/pull/1014/files#diff-1faa2c3f00938a7470735a3979b5ddbeR26

**Pull request merged**